controller/config.go

```diff
@@ -529,6 +529,15 @@ func (c Config) AutocertDNSName() string {
 	return c.asString(AutocertDNSNameKey)
 }
 
+// SetAutocertDNSName sets controller public DNS.
+func (c Config) SetAutocertDNSName(controllerDNS string) error {
```

Comment:
This is usually supplied by the user at bootstrap time via the --config argument.
Having bootstrap process overwrite it is problematic.
We could consider adding a cloudServiceDoc for the controller (and also cloudContainerDoc for the pods). This fits better with the data model and allows for when we will model controllers as applications